content/Building a SPA using Hotwire - Matteo Guerrieri.md

```diff
@@ -0,0 +1,35 @@
+## Building a SPA using Hotwire
+
+In the past years, there has been a rise in Single Page Applications ([SPA](https://en.wikipedia.org/wiki/Single-page_application)). Many of the modern web application tech stack rely on a JSON API in the backend and a frontend app modifying the DOM in the web browser, creating a fast and seamless experience.
+
+When Ruby on Rails was created, it was created to render pages on the server, so to build a SPA with Ruby on Rails running on the backend meant rendering JSON payloads as a response and let the frontend take care of the rest.
+
+But what if the frontend could update the DOM, sending HTML from the backend instead of JSON?
+
+Here is where [Hotwire](https://hotwire.dev/) comes into action!
+
+![Hotwire](images/hotwire.png)
+
+From the Hotwire webpage:
+
+> Hotwire is an alternative approach to building modern web applications without using much JavaScript by sending HTML instead of JSON over the wire. This makes for fast first-load pages, keeps template rendering on the server, and allows for a simpler, more productive development experience in any programming language, without sacrificing any of the speed or responsiveness associated with a traditional single-page application.
+
+Hotwire was created with the aim of replacing these heavy load JS SPAs with good old HTML and encourage to see the application as a set of frames that can be replaced or augmented by the HTML sent from the server.
+
+Being that said, using Hotwire means keeping the Ruby on Rails way of programming, becuase mostly uses Ruby on Rails views and partials you need to write anyway.
```

Comment:
```suggestion
That being said, using Hotwire means keeping the Ruby on Rails way of programming, because it mostly uses Ruby on Rails views and partials you need to write anyway.
```